app/javascript/components/supplemental_application/SupplementalApplicationPage.js

```diff
@@ -1,48 +1,84 @@
 import React from 'react'
+import { cloneDeep } from 'lodash'
 
-import SupplementalApplicationContainer from './SupplementalApplicationContainer'
 import appPaths from '~/utils/appPaths'
 import mapProps from '~/utils/mapProps'
-import CardLayout from '../layouts/CardLayout'
 import { mapApplication, mapFieldUpdateComment } from '~/components/mappers/soqlToDomain'
 import { updateApplicationAction } from './actions'
 import { mapList } from '~/components/mappers/utils'
+import CardLayout from '../layouts/CardLayout'
+import SupplementalApplicationContainer from './SupplementalApplicationContainer'
 
-const SupplementalApplicationPage = ({ application, statusHistory, formFields, onSubmit, fileBaseUrl }) => {
-  const pageHeader = {
-    title: `${application.name}: ${application.applicant.name}`,
-    breadcrumbs: [
-      { title: 'Lease Ups', link: '/lease-ups' },
-      { title: application.listing.name, link: appPaths.toListingLeaseUps(application.listing.id) },
-      { title: application.name, link: '#' }
-    ]
+class SupplementalApplicationPage extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      persistedApplication: cloneDeep(props.application),
+      application: props.application
```

Comment:
setting the state equal to the props seems like pretty weird behavior. then we have this.state.application, this.state.persistedApplication, this.props.application, and this.props.formApi.values all holding the same thing, right?

Reply:
yes, kind of, we need this because of a complicated behavior. setting state based on props is usually not recommended but we do not have a global state or state management and this is the root component so we are safe. I was kind of trying to make this page the store. We can talk on stand up if there is any other way to solve this problem